### PR TITLE
Restructure commands in core

### PR DESCRIPTION
**Pull request by eggplantzzz**
This PR aims to create a regular and consistent structure for Truffle command files.

This PR creates a folder for each command in which the `run` method for each command is separate from what I am calling the "meta stuff". This "meta stuff" includes info for the help as well as material that yargs uses to determine options and their types. Enjoy!

### CODE REVIEW

**Comment by eggplantzzz** [comment]
I'm glad I got PR #4444

**Review by kevinweaver — APPROVED** [review]
👍 nice refactor!  I spent a few minutes scanning through these, but with passing tests, LGTM

**Comment by cds-amal** [comment]
While testing this PR I noticed a difference in behavior between production for truffle develop for the following smoke test where prod has an extra key (`_`) in the output. :man_shrugging: 

``` console
$ truffle develop
truffle(develop) Object.keys(this)
```


<details>
<summary>PR branch</summary>

```
truffle(develop)> Object.keys(this)
[
  'global',             'clearInterval',
  'clearTimeout',       'setInterval',
  'setTimeout',         'queueMicrotask',
  'clearImmediate',     'setImmediate',
  'crypto',             'regeneratorRuntime',
  '__core-js_shared__', 'web3',
  'interfaceAdapter',   'accounts',
  'ConvertLib',         'MetaCoin',
  'Migrations'
]
truffle(develop)> .exit
```

</details>

<details>
<summary>Truffle v5.4.22</summary>

```
truffle(develop)> Object.keys(this)
[
  'global',             'clearInterval',
  'clearTimeout',       'setInterval',
  'setTimeout',         'queueMicrotask',
  'clearImmediate',     'setImmediate',
  'crypto',             'regeneratorRuntime',
  '__core-js_shared__', '_',
  'web3',               'interfaceAdapter',
  'accounts',           'ConvertLib',
  'MetaCoin',           'Migrations'
]
truffle(develop)>
```

</details>

**Comment by eggplantzzz** [comment]
Huh, that's weird. Other than that @cds-amal, did all the smoke tests look good?

I wonder where that _ came from...

**Comment by haltman-at** [comment]
My understanding is that the `_` key contains any arguments not tied to a specific option.  We use it in the debugger CLI, for instance, for getting the transaction hash.  Does the debugger CLI still work on this branch?

**Comment by eggplantzzz** [comment]
On this branch I tested by opening a Truffle develop session, migrating, and then using `debug <txHash>` to debug a deployment. It worked as expected.

File: packages/core/lib/commands/migrate/run.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        return await Migrate.run(config);
+      } else {
+        config.logger.log("Network up to date.");
+        return;
+      }
+    }
+  }
+};
+
+const prepareConfigForRealMigrations = async function (buildDir, options) {
```
Comment (cds-amal) [inline packages/core/lib/commands/migrate/run.js:84]: `determineDryRunSettings` was extracted to its own file. Is there a reason for not doing the same for this?

File: packages/truffle/index.js [hunk 2]
```diff
@@ -0,0 +1 @@
+console.log(__dirname);
```
Comment (cds-amal) [inline packages/truffle/index.js:1]: Why's this necessary?

**Review by cds-amal — COMMENTED** [review]
This looks good but I want to run a couple more smoke tests and track down that `_` issue before I can approve

File: packages/core/lib/commands/migrate/run.js (hunk 1 above)
Comment (eggplantzzz) [reply packages/core/lib/commands/migrate/run.js:84]: I like it!

File: packages/truffle/index.js (hunk 2 above)
Comment (eggplantzzz) [reply packages/truffle/index.js:1]: oof, honestly I don't know where this came from. Good catch!

File: packages/core/lib/commands/console/run.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+module.exports = async function (options) {
+  const Config = require("@truffle/config");
+  const Console = require("../../console");
+  const { Environment } = require("@truffle/environment");
+
+  const config = Config.detect(options);
+
+  // This require a smell?
```
[the remark this answers is not in the document]
Comment (eggplantzzz) [reply packages/core/lib/commands/console/run.js]: deleted!

**Comment by cds-amal** [comment]
`_` occurs consistently in the bundled version. not a problem . Lets get this in @eggplantzzz !

**Review by cds-amal — APPROVED** [review]
Nice work @eggplantzzz! glad to get this in.

File: packages/core/lib/commands/db/meta.js [hunk 4]
```diff
@@ -13,17 +13,20 @@ module.exports = {
...
+      ...serveCommand.run,
+      ...serveCommand.meta
+    }).demandCommand();
   },
 
   subCommands: {
     serve: {
       help: serveCommand.help,
-      description: serveCommand.description
+      description: serveCommand.meta
```
Comment (cds-amal) [inline packages/core/lib/commands/db/meta.js:25]: good catch! Missed that one.

**Pull request merged**